app/ReaderWriter.cpp

```diff
@@ -1,13 +1,27 @@
+/*@
+ *@file       ImageInput.cpp
+ *@author     Umang Rastogi
+ *@brief      Source file to controls the actual source implementation of reader
+ *@copyright  2019 Umang Rastogi
+ */
+
 #include <ReaderWriter.hpp>
 
-cv::Mat ReaderWriter::read(std::string imagePath) {
-  return cv::Mat();
+ReaderWriter::ReaderWriter {}
+
+cv::Mat ReaderWriter::read(std::String imagePath) {
+  // convert the image file into Mat format
+  return cv::imread(imagePath);
```

Comment:
Throw exception if image not found. Check if Mat returned is NULL.

Reply:
Added exception handling